### PR TITLE
add method custom Paint

### PR DESCRIPTION
**Pull request by ttpho**
### Overview 
I want to add method to custom paint for `CircularSeekBar`

### Use 
Screenshot_20220722_181748

```kotlin

        val dashWidth = convertDpToPixel(16.0f, context.resources) // 16 dp to pixel
        val dashGap = dashWidth / 2.0f
        circularSeekBar?.customCirclePaint {
            it.pathEffect = DashPathEffect(floatArrayOf(dashWidth, dashGap), 0.0f)
        }

```

### CODE REVIEW

File: circularSeekBar/src/main/java/me/tankery/lib/circularseekbar/CircularSeekBar.kt [hunk 1]
```diff
@@ -1144,6 +1144,39 @@ class CircularSeekBar @JvmOverloads constructor(
         recalculateAll()
     }
 
+    private fun customPaint(paint: Paint, block: (Paint) -> Unit) {
```
Comment (tankery) [inline circularSeekBar/src/main/java/me/tankery/lib/circularseekbar/CircularSeekBar.kt:1147]: To show a more complete meaning for these methods, do you think adding a verb prefix for the naming would make more sense? Such as "applyCustomXXX".

File: circularSeekBar/src/main/java/me/tankery/lib/circularseekbar/CircularSeekBar.kt [hunk 2]
```diff
@@ -1144,6 +1144,39 @@ class CircularSeekBar @JvmOverloads constructor(
...
+        block.invoke(paint)
+        invalidate()
+    }
+
+    fun customCircleFillPaint(block: (Paint) -> Unit) {
```
Comment (tankery) [inline circularSeekBar/src/main/java/me/tankery/lib/circularseekbar/CircularSeekBar.kt:1152]: Could you adding some Java docs for these public methods to explain what it does?

File: circularSeekBar/src/main/java/me/tankery/lib/circularseekbar/CircularSeekBar.kt [hunk 3]
```diff
@@ -1144,6 +1144,39 @@ class CircularSeekBar @JvmOverloads constructor(
...
+    }
+
+    fun customPointerHaloPaint(block: (Paint) -> Unit) {
+        customPaint(pointerHaloPaint, block)
+    }
+
+    fun customPointerHaloBorderPaint(block: (Paint) -> Unit) {
+        customPaint(pointerHaloBorderPaint, block)
+    }
+
```
Comment (tankery) [inline circularSeekBar/src/main/java/me/tankery/lib/circularseekbar/CircularSeekBar.kt:1179]: nit: Can we add these method after the `setOnSeekBarChangeListener`.

**Comment by tankery** [comment]
Hi, @ttpho , Thanks for the PR and sorry for the late reply. The idea of setting custom painters looks great, could provide a more flexible choice for developers. Just added some comments, could you please take a look? Thanks again for the contribution!

**Comment by ttpho** [comment]
@tankery Thank you for your review. Please help me check my [commit](https://github.com/tankery/CircularSeekBar/pull/50/commits/de0985f232561b52329d7f0d43928e704e8a220e).

**Comment by tankery** [comment]
Thanks for the contribution. This new method has been included in the version 1.4.1 :)

**Pull request merged**